pandas/tests/test_common.py

```diff
@@ -217,13 +220,23 @@ def test_standardize_mapping():
     Series(100 * [0.123456, 0.234567, 0.567567], name='X')])
 @pytest.mark.parametrize('method', ['to_pickle', 'to_json', 'to_csv'])
 def test_compression_size(obj, method, compression_only):
-
-    with tm.ensure_clean() as filename:
-        getattr(obj, method)(filename, compression=compression_only)
-        compressed = os.path.getsize(filename)
-        getattr(obj, method)(filename, compression=None)
-        uncompressed = os.path.getsize(filename)
-        assert uncompressed > compressed
+    """
+    Tests that compression is occurring by comparing to the bytes on disk of
+    the uncompressed file.
+    """
+    extension = _compression_to_extension[compression_only]
```

Comment:
Would prefer to keep this test unchanged and ensure that the test above is responsible for `infer` behavior

Reply:
I think the paremetrization of this test is ideal. Currently this tests compression X to_method X dataframe_or_series. Luckily this exhaustive parametrization caught that I hadn't switched the default for Series.to_csv. Also note that the functional changes to this test are minimal (just adding getting the size of an inferred file). The rest of the changes help readability.

Were we to leave `test_compression_size` unmodified, I'd want to repeat the parametrization. Is there a dataframe_or_series fixture?